src/Moryx.ControlSystem/VisualInstructions/IVisualInstructor.cs

```diff
@@ -14,34 +15,22 @@ public interface IVisualInstructor : IResource
         /// Only display these instructions
         /// Have to be cleared with the <see cref="Clear"/> method
         /// </summary>
-        long Display(string sender, VisualInstruction[] instructions);
+        long Display(ActiveInstruction instruction);
 
         /// <summary>
         /// Only display these instructions
         /// Instruction will automatically cleared after the given time
         /// </summary>
-        void Display(string sender, VisualInstruction[] instructions, int autoClearMs);
+        void Display(ActiveInstruction instruction, int autoClearMs);
 
         /// <summary>
-        /// Execute these instructions based on the given activity and report the result on completion
-        /// Can (but must not) be cleared with the <see cref="Clear"/> method
+        /// Execute these instructions based on the given activity and report the result and inputs on completion
```

Comment:
From an outside perspective: This method signature says nothing about an activity, also the phrase "report the results and inputs" will probably be forgotten to be changed when the ActiveInstructionResult changes in the future (and then be incomplete or even misleading ).